src/Microsoft.Dnx.Compilation.CSharp/CompilationContext.cs

```diff
@@ -52,6 +55,8 @@ public class CompilationContext
 
         public IList<ICompileModule> Modules { get; }
 
+        public ImmutableArray<DiagnosticAnalyzer> DiagnosticAnalyzers { get; set; }
```

Comment:
We don't use ImmutableArray anywhere but I'm vaguely interested in doing it more. Do you know anything about the performance impact?